intro-to-R-tidyverse/03-intro_to_r_tidyverse_exercise.Rmd

```diff
@@ -0,0 +1,189 @@
+
+# reading gene expression matrix
+gene_expression <- 
+```
+
+### Cleaning metadata
+First, we are going to clean the metadata using filtering, renaming, reordering 
+and we will write our newly cleaned metadata into a new file. 
+
+#### Inspect the metadata
+Now that you have read the metadata into the variable `metadata`, explore what 
+the metadata looks like. 
+
+```{r}
+# explore the metadata
+
+```
+
+#### Filtering and trimming
+Next, we're going to do some manipulations on the metadata to retain only the 
+information we are interested in. 
+Below we have done some manipulations on the metadata. 
+Take a look the steps we performed and then perform the same manipulations but 
+by using pipes `%>%` and storing the result in `metadata_clean`. 
+
+```{r}
+# manipulations w/o pipes
+metadata_nopipe <- metadata
+metadata_nopipe <- filter(metadata_nopipe, refinebio_age < 18)
+metadata_nopipe <- select(metadata_nopipe, refinebio_accession_code, 
+                          refinebio_age, refinebio_sex, histology)
+metadata_nopipe <- arrange(metadata_nopipe, refinebio_age)
+```
+
+It's your turn! 
+
+```{r}
+# now you try it, with pipes %>%
+metadata_clean <- 
+  
+```
+
+Run the code chunk below to check if you got the same results. 
+
+```{r}
+all.equal(metadata_nopipe, metadata_clean)
+```
+
+#### Rename columns
+The column names `refinebio_accession_code`, `refinebio_age`, `refinebio_sex` 
+are too unwieldy. 
+Let's rename them to something more manageable like `sample_id`, `age`, and 
+`sex` respectively. 
+
+To do this, tidyverse has a function `dplyr::rename`. 
+Use the help window to understand how you can use this function. 
+Store the renamed metadata back into the variable `metadata_clean`. 
+(Feel free to google for solutions as well!)
+
+```{r}
+# use the help window to learn how you can use the rename function
+metadata_clean <- 
+
+```
+
+Use `head` function check if the column names changed.
+
+```{r}
+
+```
+
+### Summarize metadata
+Now, let's calculate the mean age by sex, mean age by histology, and counts of 
+each tissue by sex. 
+
+If you get stuck at any point, feel free to refer to the help window or google 
+for solutions.
+
+```{r}
+# calculate mean age by sex
+
+```
+
+```{r}
+# calculate mean age by histology
+
+```
+
+```{r}
+# calculate how many of each tissue type exists for each sex
+
+```
+
+### Select the samples in the gene expression matrix
+Now, select the samples in the gene expression matrix so that it has the same 
+samples as `metadata_clean`. 
+(Remember we read the gene expression matrix into `gene_expression`)
+
+```{r}
+# select the gene expression matrix by samples in the cleaned metadata
+cleaned_gene_expression <- 
+```
+
+### Write to files
+
+#### Write to cleaned metadata and gene expression matrix to file
+Last step, write the cleaned metadata into `metadata_GSE49243_cleaned.tsv` and 
+cleaned gene expression 
+matrix into `GSE49243_cleaned.tsv` the results directory (which we saved as 
```

Comment:
A lot of this sentence can be up on the next line after `cleaned gene expression`